### PR TITLE
[k]  index and sorting category ops fix

### PR DESCRIPTION
**Pull request by RobertLuobo**
### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/runtime/backend/_kunlunxin/ops/masked_select.py [hunk 1]
```diff
@@ -24,6 +24,29 @@
...
+except Exception:  # pragma: no cover
+    logger.warning("masked_select: unable to acquire raw native nonzero kernel")
+    _NATIVE_NONZERO = None
+    _VENDOR_KS = None
+
+
+def _native_nonzero(x):
+    if _NATIVE_NONZERO is not None:
+        return _NATIVE_NONZERO.call_boxed(_VENDOR_KS, x)
+    return x.nonzero()
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/masked_select.py]: This explicitly bypasses FlagGems by fetching and invoking the raw native nonzero kernel. Please replace this path with a Triton/FlagGems or Kunlunxin backend implementation instead of calling the native ATen kernel directly.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/segment_reduce.py [hunk 2]
```diff
@@ -0,0 +1,1301 @@
...
+        return torch.squeeze(reshaped, dim=reduce_dim)
+    if reduce == "sum":
+        return torch.sum(reshaped, dim=reduce_dim)
+    if reduce == "mean":
+        return torch.mean(reshaped, dim=reduce_dim)
+    if reduce == "max":
+        return torch.amax(reshaped, dim=reduce_dim)
+    if reduce == "min":
+        return torch.amin(reshaped, dim=reduce_dim)
+    return torch.prod(reshaped, dim=reduce_dim)
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/segment_reduce.py]: For uniform segments longer than 256, this path directly calls PyTorch reduction operators from the host. Please use a Triton implementation or explicitly route through existing FlagGems operators.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/bucketize.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tle.program_id(0)
+    offsets = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = offsets < n_elements
+    v = tl.load(inp_ptr + offsets, mask=mask, other=0.0).to(tl.float32)
+
+    idx = tl.zeros([BLOCK_SIZE], dtype=tl.int32)
+    for i in tl.static_range(N_BOUNDARIES):
+        b = tl.load(boundaries_ptr + i).to(tl.float32)
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/bucketize.py:48]: Converting both inputs and boundaries to tl.float32 changes the result for large int64 and closely spaced FP64 values. Please preserve the comparison dtype and add parity tests around 2**24 and adjacent FP64 boundaries.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/index_copy_.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    out_coord2 = tl.where(dim == 2, indices, coord2)
+    src_offset = coord0 * src_stride0 + coord1 * src_stride1 + coord2 * src_stride2
+    out_offset = (
+        out_coord0 * inp_stride0 + out_coord1 * inp_stride1 + out_coord2 * inp_stride2
+    )
+    src_values = tl.load(src + src_offset, mask=mask)
+    tl.store(inp + out_offset, src_values, mask=mask)
+
+
+def _validate(inp, dim, index, src):
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/index_copy_.py:109]: Please normalize dim before _validate. Currently the shape check never recognizes a negative target dimension, so valid calls where src.size(-1) != self.size(-1) are rejected.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/segment_reduce.py (hunk 2 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/segment_reduce.py]: It is same as src/flag_gems/ops/segment_reduce.py
image

File: src/flag_gems/runtime/backend/_kunlunxin/ops/bucketize.py (hunk 3 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/bucketize.py:48]: But the testcase all pass.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/index_copy_.py (hunk 4 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/index_copy_.py:109]: Image

File: src/flag_gems/runtime/backend/_kunlunxin/ops/segment_reduce.py (hunk 2 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/segment_reduce.py]: DONE

Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/segment_reduce.py]: DONE

**Review by Caeruleann — APPROVED** [review]
